### PR TITLE
AC-948 Migrated Logs Package from MVP to MVVM

### PR DESCRIPTION
**Pull request by shivamsawlani02**
## Description of what I changed
Migrated Logs Package from MVP to MVVM. 
Removed the Contract and Presenter Classes and added a ViewModel class.

## Issue I worked on
JIRA Issue: https://issues.openmrs.org/browse/AC-948

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by shivamsawlani02** [comment]
@LuGO0 @rishabh-997 please review.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/logs/LogsFragment.kt [hunk 1]
```diff
@@ -21,31 +21,47 @@ import android.view.LayoutInflater
...
+    private fun attachLogsToTextView() {
+        mViewModel.getLogs()
+        mViewModel.logsText.observe(viewLifecycleOwner, Observer { updatedLogTexts ->
+            binding.tvLogs.text = updatedLogTexts
+        })
+    }
+
+    private fun fabCopyAll() {
+        mViewModel.getLogs()
+        mViewModel.logsText.observe(viewLifecycleOwner, Observer { updatedLogTexts ->
```
Comment (HerbertYiga) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/logs/LogsFragment.kt:60]: thanks @LuGO0  how do things look to you on this entire pr?

**Comment by LuGO0** [comment]
@shivamsawlani02 we need to start from the bottoms up I think making a base class structure before adding in classes on top of it. I think this wont work.

**Comment by shivamsawlani02** [comment]
> @shivamsawlani02 we need to start from the bottoms up I think making a base class structure before adding in classes on top of it. I think this wont work.

Okay, @LuGO0. I will create a new issue in JIRA for creating Base classes for Fragments and ViewModels. There isn't any change in Activities, so I think we don't need a new BaseActivity. So, will creating a new BaseFragment and a BaseViewModel be sufficient, or do we need to do something else also?

**Comment by LuGO0** [comment]
yes I think so then I will have to keep this Issue on hold uptill then since it will be based on those base classes rather than be independent, and you might want to have a look we have a BaseFragment as well if that helps.

**Comment by shivamsawlani02** [comment]
> yes I think so then I will have to keep this Issue on hold uptill then since it will be based on those base classes rather than be independent, and you might want to have a look we have a BaseFragment as well if that helps.

Yes, we can keep this on hold till base classes are created.

**Comment by LuGO0** [comment]
closing this for now.

**Pull request closed without merging**